WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentsViewModel.kt

```diff
@@ -154,30 +176,37 @@ class NewSiteCreationSegmentsViewModel
             items.add(segment)
         }
     }
+}
 
-    data class UiState(
-        val showError: Boolean,
-        val showContent: Boolean,
-        val items: List<ItemUiState>
-    )
+interface SegmentsUiState
```

Comment:
This should be a `sealed class` since there are only 2 possible values for it and they are mutually exclusive.

Reply:
With my other comments combined, here is the approach I'd have taken:

```
sealed class SegmentsUiState {
    data class SegmentsContentUiState(val items: List<SegmentsItemUiState>) : SegmentsUiState()
    sealed class SegmentsErrorUiState constructor(
        val titleResId: Int,
        val subtitleResId: Int? = null
    ) : SegmentsUiState() {
        object genericErrorUiState : SegmentsErrorUiState(
                R.string.site_creation_error_generic_title,
                R.string.site_creation_error_generic_subtitle
        )

        object connectionErrorUiState : SegmentsErrorUiState(R.string.site_creation_error_connection_title)
    }
}
```

Reply:
👍